### DISCUSSION TITLE
Kits and PersistentDataHolder

### DISCUSSION THREAD

**Discussion by PancakeCatsss** [EssentialsX/Essentials#3984] [Ideas and Feature Suggestions] [answered]
Kits currently do not save persistent data on items so custom items that use the PersistentDataHolder cannot be given in the kits. Hoping this could be something to add in the future.

**Comment by mdcfe** [comment, marked as answer]
Support for serializing custom data stored with the `PersistentDataHolder` API isn't planned for 2.x at the moment.

In the meantime, plugins can integrate with the [`ItemResolver` APIs](https://github.com/EssentialsX/Essentials/blob/2.x/Essentials/src/main/java/net/ess3/api/IItemDb.java) to allow spawning and serialization of custom items in kits.
